components/centraldashboard/public/components/resource-chart_test.js

```diff
@@ -0,0 +1,215 @@
+    it('Draws chart with top 5 time-series for podmem', async () => {
+        resourceChart.metric = 'podmem';
+        resourceChart.interval = 'Last30m';
+        const metrics = [
+            {timestamp: 1558621807, label: 'node-1', value: 1},
+            {timestamp: 1558621867, label: 'node-1', value: 1},
+            {timestamp: 1558621927, label: 'node-1', value: 1},
+            {timestamp: 1558621807, label: 'node-2', value: 2},
+            {timestamp: 1558621867, label: 'node-2', value: 2},
+            {timestamp: 1558621927, label: 'node-2', value: 2},
+            {timestamp: 1558621807, label: 'node-3', value: 3},
+            {timestamp: 1558621867, label: 'node-3', value: 3},
+            {timestamp: 1558621927, label: 'node-3', value: 3},
+            {timestamp: 1558621807, label: 'node-4', value: 4},
+            {timestamp: 1558621867, label: 'node-4', value: 4},
+            {timestamp: 1558621927, label: 'node-4', value: 4},
+            {timestamp: 1558621807, label: 'node-5', value: 5},
+            {timestamp: 1558621867, label: 'node-5', value: 5},
+            {timestamp: 1558621927, label: 'node-5', value: 5},
+            {timestamp: 1558621807, label: 'node-6', value: 6},
+            {timestamp: 1558621867, label: 'node-6', value: 6},
+            {timestamp: 1558621927, label: 'node-6', value: 6},
+            {timestamp: 1558621807, label: 'node-7', value: 7},
+            {timestamp: 1558621867, label: 'node-7', value: 7},
+            {timestamp: 1558621927, label: 'node-7', value: 7},
+            {timestamp: 1558621807, label: 'node-8', value: 8},
+            {timestamp: 1558621867, label: 'node-8', value: 8},
+            {timestamp: 1558621927, label: 'node-8', value: 8},
+            {timestamp: 1558621807, label: 'node-9', value: 9},
+            {timestamp: 1558621867, label: 'node-9', value: 9},
+            {timestamp: 1558621927, label: 'node-9', value: 9},
+            {timestamp: 1558621807, label: 'node-10', value: 10},
+            {timestamp: 1558621867, label: 'node-10', value: 10},
+            {timestamp: 1558621927, label: 'node-10', value: 10},
+        ];
+        const responsePromise = mockRequest(resourceChart, {
+            status: 200,
+            responseText: JSON.stringify(metrics),
+        }, false, '/api/metrics/podmem?interval=Last30m');
+        await responsePromise;
+        flush();
+
+        expect(resourceChart.shadowRoot.getElementById('interval')
+            .selectedItem.innerText).toBe('30m');
+        // Validate chart properties directly
+        expect(resourceChart._chart.data.datasets.length).toBe(5);
+        expect(resourceChart._chart.options.scales.yAxes[0].ticks
+            .stepSize).toBeUndefined();
+        expect(resourceChart._chart.options.legend.display).toBe(false);
+
+        const [ds1, ds2, ds3, ds4, ds5] = resourceChart._chart.data.datasets;
+        expect(ds1.label).toBe('node-10');
+        expect(ds1.backgroundColor).toBe('#1e88e5');
```

Comment:
Should we be testing background colors? Assuming we want to stick to the addage:

> Test Behavior, Not Implementation **([tott](https://testing.googleblog.com/2013/08/testing-on-toilet-test-behavior-not.html))**

Reply:
Normally I wouldn't want to do so, but in this case since we are implementing the charts specification and we don't have an easy way to verify the content of the chart in test, it seemed like a good way to validate that each series has a unique color that matches expectations.

Reply:
Agree we want the cards to left align in the absence of cards in the middle. So anytime there's content missing to the left, move the card left.. and any time there's content missing below, move the card up.